### PR TITLE
Tooling API: Support jdkName from idea module model

### PR DESCRIPTION
**Pull request by ingokegel**
The information specified by

```gradle
idea {
    module {
	    jdkName = "My JDK"
	}
}
```

is not made available by the tooling API. In order to fix

https://youtrack.jetbrains.com/issue/IDEA-138526

this change is required first. The language level is already handled automatically in recent versions of IDEA. However, if modules have different language levels, the project JDK is not suitable for them and you have to be able to set a module-specific JDK.

- [ x] Reviewed the [Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md).
- [x ] Signed the [Gradle CLA](http://gradle.org/contributor-license-agreement/).
- [ x] Ensured that basic checks pass: `./gradlew quickCheck`

### CODE REVIEW

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/idea/DefaultIdeaModule.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
     private IdeaCompilerOutput compilerOutput;
 
     private DefaultIdeaJavaLanguageSettings javaLanguageSettings;
+    private String jdkName;
```
Comment (oehme) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/idea/DefaultIdeaModule.java]: Shouldn't this be on the Java language settings?

**Comment by ingokegel** [comment]
The problem is that DefaultIdeaProject already has a jdkName property, so
putting it into IdeaJavaLanguageSettings would duplicate this information.
Is that OK or should the related top-level properties in DefaultIdeaProject
be deprecated?

2016-12-09 23:26 GMT+01:00 Stefan Oehme <notifications@github.com>:

**Comment by oehme** [comment]
Sorry you are right, nevermind my last comment :)

**Comment by oehme** [comment]
The change looks good, would you mind adding a cross-version test to make sure it doesn't get broken in the future? See `ToolingApiIdeaModelCrossVersionSpec` for examples.

**Comment by ingokegel** [comment]
OK, I have added it. If I understand correctly, this would now go into 3.4.

**Comment by oehme** [comment]
Excellent, thank you Ingo! I'm out this week, but will merge this early next week.

**Comment by eriwen** [comment]
@ingokegel Yes, it'll go in the next version of Gradle once merged. 3.3 is locked down for everything except critical bug fixes.

**Comment by eriwen** [comment]
@ingokegel This was rebased and applied with 8e02bc268f6478773ca6daf49bea9aed14c5d1eb — thank you for contributing!

**Pull request closed without merging**